#41 Create "About Us" Page Programmatically

lrowang opened the pull request:
Hi @lhridley -

PR for #41.

I've made some progress creating the Page Node programmatically via `hook_update`, but I am running into 2 issues I could use some help with:

1. I can create my "about us" page, but **I can't figure out why the path has an extra value**. The page is showing up as `/content/about-us` instead of as `/about-us`. And actually, the first time I ran it _it did_ show us as `/about-us` but hasn't since.  I delete the content via the UI inbetween tries, but it's still producing the extra path variable. 

**pic1**
![cs 41-path](https://cloud.githubusercontent.com/assets/19154706/19062288/a8c44326-89ad-11e6-9d08-8df13c1d0d86.png)

<br>
2. **I'd like to be able to add the 3 paragraphs of "content" in a more modular fashion**, rather than just pasting them into the code body of `hook_update`. I tried using the function `file_get_contents` but it's not quite working the way I expected (and I can't figure it out). You can see that I can create content directly by pasting (pic 2), but if I try running it using `file_get_contents` I get nothing on the page (pic 3).

**pic2**
![cs 41-yestext](https://cloud.githubusercontent.com/assets/19154706/19062226/fd53afc2-89ac-11e6-859e-cdd5ed8f51b8.png)

**pic3**
![cs 41-notext](https://cloud.githubusercontent.com/assets/19154706/19062205/c5e5451e-89ac-11e6-87be-bc5123817e5b.png)

lhridley commented:
(1) You've got a conflict because you are reusing a hook  update number -- you used 7119 to enable the "contact_contact_us" module.  Please renumber this hook update to 7120.

lrowang commented:
Hi @lhridley -

I've incorporated all the comments above and in addition:
- Created a helper function for the code so it can be placed in both `hook_install` and `hook_udpate` more cleanly.
- Created a simple behat test scenario to check the "About Us" page is available and displaying the correct content
- I **did not** update the Basic Page's pathauto pattern, and kept it as `page/[node:title]` because that's what was specified in #12. So the "About Us" page's path is `/page/about-us`.
- Added that additional modification to the `behat.yml` file that clears the gherkin cache.

Local tests passed and PR is ready for official review!

P.s. Also it was a great opportunity to do my first rebase! I used a combo of terminal, phpstorm, and source tree!
